src/compile/mark/valueref.ts

```diff
@@ -83,18 +75,25 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   if (channelDef) {
     /* istanbul ignore else */
     if (isFieldDef(channelDef)) {
+      if (isBinScale(scale.type)) {
+        if (contains(['x', 'y', 'x2', 'y2', 'size'], channel)) {
```

Comment:
Based on [this revision](https://github.com/vega/vega-lite/pull/2009/files#r104353272), we might not want to call `isBinScale` here, but instead add special case inside two branches (discrete/continuous) below.